packages/skeleton/src/lib/index.ts

```diff
@@ -14,6 +15,17 @@ export type { PopupSettings } from './utilities/Popup/types.js';
 // This type alias is to identify CSS classes within component props, which enables Tailwind IntelliSense
 export type CssClasses = string;
 
+// Transition
+interface AdditionalTransitionProps {
+	ignoreReducedMotion?: boolean;
+}
+export type Transition = (node: Element, params?: any) => TransitionConfig;
+export type TransitionParams<T extends Transition> = Parameters<T>[1] & AdditionalTransitionProps;
+export interface TransitionSettings<T extends Transition> {
+	transition: T;
+	params?: TransitionParams<T>;
+}
+
```

Comment:
Any suggestions on where and how we name this?

Reply:
It's an awkward one since it can be used in places like in both `lib/components/` and `lib/utilities/`. The easiest would be to just move all the Transition types along with the `CssClasses` type alias into `src/lib/types.ts`. We can use this file to stick all the types that are used rather often throughout the project.

A bit icky, but it keeps our entry point clean.

Reply:
Hmm, I don't like the idea of dropping it in the component or utilities directories given scope can be shared between both.

I believe `/lib/types.ts` was my original proposal for this, but it somehow got merged into `index.ts`. We're likely going to continue having "global" types added over time, so best to decide where we want those to live sooner than later.

I'd say we start there and feel it out.